tutorials/nginx-reverse-proxy-docker.md

```diff
@@ -0,0 +1,392 @@
+    http://a.example.com, or http://b.example.com.
+
+1.  Stop the container.
+
+        docker stop site-a
+
+1.  Remove the container.
+
+        docker rm site-a
+
+You now have a static website which you can run in a Docker container. Repeat
+these steps to get a second site set up.
+
+1.  Create directory for "site B."
+
+        cd
+        mkdir site-b
+        cd site-b
+
+1.  Configure a Docker image for "site B."
+
+        echo "FROM kyma/docker-nginx
+        COPY src/ /var/www
+        CMD 'nginx'" > Dockerfile
+
+1.  Make the static website pages.
+
+        mkdir src
+        echo "Hello from site B" > src/index.html
+
+1.  Build the Docker image. This time will be faster because the base image is
+    cached.
+
+        docker build -t site-b .
+
+1.  Run the Docker container.
+
+        docker run -d --name site-b -p 80:80 site-b
+
+1.  View the running website at http://EXTERNAL_IP_ADDRESS,
+    http://a.example.com, or http://b.example.com.
+
+1.  Stop the container.
+
+        docker stop site-b
+
+1.  Remove the container.
+
+        docker rm site-b
+
+Now you have two web apps, but since they both listen to port 80, you can't run
+them both at the same time. Also, both apps respond to requests to
+http://a.example.com and http://b.example.com. Ideally "site A" would listen to
+http://a.example.com and "site B" would listen to http://b.example.com.
+
+## Setting up the reverse proxy
+
+To have the separate apps respond only to their respective hosts, you'll
+use a [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy). This
+tutorial uses the [nginx-proxy Docker
+container](https://github.com/jwilder/nginx-proxy).
+
+1.  Run the reverse proxy.
+
+        docker run -d \
+            --name nginx-proxy \
+            -p 80:80 \
+            -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy
+
+1.  Start the container for site A, specifying the domain name in the
+    `VIRTUAL_HOST` variable.
+
+        docker run -d --name site-a -e VIRTUAL_HOST=a.example.com site-a
+
+1.  Check out your website at http://a.example.com.
+1.  With site A still running, start the container for site B.
+
+        docker run -d --name site-b -e VIRTUAL_HOST=b.example.com site-b
+
+1.  Check out site B at http://b.example.com.
+
+Congratulations, you are running multiple apps on the same host using
+Docker and an [nginx reverse proxy](https://github.com/jwilder/nginx-proxy).
+You can do even better, though.
+
+Plain HTTP is not secure. It is not encrypted and is vulnerable to
+[man-in-the-middle
+attacks](https://en.wikipedia.org/wiki/Man-in-the-middle_attack). In the next
+step, you'll add support for the HTTPS protocol.
+
+1.  Stop the containers.
+
+        docker stop site-a
+        docker stop site-b
+        docker stop nginx-proxy
+
+1.  Remove the containers.
+
+        docker rm site-a
+        docker rm site-b
+        docker rm nginx-proxy
+
+## Setting up HTTPS with Let's Encrypt
```

Comment:
Thanks Jason.

Okay. I'm still not sure we should split this up. If you look closely, it's all the same steps as the unencrypted HTTP. I'd prefer to show people an easy way to securely host their apps.

Maybe I should delete the plain HTTP steps since they are so similar and just skip to this?